processor/src/advice/mod.rs

```diff
@@ -60,6 +62,32 @@ impl AdviceProvider {
         self.tape.push(value);
     }
 
+    /// Removes vector from tapes map and returns it.
+    pub fn get_values(&mut self, key: Word) -> Option<Vec<Felt>> {
+        self.tapes.remove(&key.into_bytes())
+    }
+
+    /// Writes the provided vector at the head of the advice tape.
+    ///
+    /// # Errors
+    /// Returns an error if the value for this key was not found in the tapes map
+    pub fn write_tape_many(
+        &mut self,
+        map_value: Option<Vec<Felt>>,
+        key: Word,
+    ) -> Result<(), ExecutionError> {
```

Comment:
This method should write to the advice tape - not to the key-value map. The signature would be:
```Rust
pub fn write_tape_many(&mut self, values: &[Felt]);
```